pyrobosim/pyrobosim/utils/general.py

```diff
@@ -75,32 +76,68 @@ def get(self, category):
         """
         return self.data.get(category, None)
 
+    def update(self, new_data, source=None):
+        """
+        Update the metadata with new data and add a source of this data.
+
+        :param new_data: New metadata to add.
+        :type new_data: dict
+        :param source: Source of new metadata
+        :type source: str, optional
+        """
+        if source is not None:
+            self.sources[self.metadata_type].append(source)
+
+        for key, value in new_data.items():
+            if key in self.data and self.data[key] != value:
+                raise MetadataConflictException(key, self.data[key], value, source)
+            self.data[key] = value
```

Comment:
I'm actually not sure whether we want to raise an exception, or simply overwrite the metadata with the new file. I'm OK leaving it as is for now.